Feat/#56 캘린더 주간 API 기본구조

hooni0918 opened the pull request:
## 🔗 연결된 이슈
- Closed: #56 

## 📄 작업 내용
- 캘린더 DTO 및 관련 기능API를 붙였습니다.
- 이전에 논의햇던 Alarmofire를 Moya처럼 사용하자는것을 수용하여 구조를 짯었으나 기존 작업중인 코드들과 추후 코드 일을때 너무 파편화될게 우려되어 기존 Alarmofire를 Alarmofire답게 사용햇습니다 ~~(내코드돌려줘)~~'
- API가 아직 불안정한 관계로 지금PR은 그냥 기본 구조 잡기용으로 생각해주시고 해당 PR머지후 남은 UI 작업 -> 나머지 API구조잡기 -> API 연결최종_진짜최종 순으로 가야할것같아요ㅠ

## 💻 주요 코드 설명
기본 코드라고 생각해서 딱히 설명은 하지 않겟습니다?




### ✔️ CI Completed
- ✅ Lint: Completed
- ✅ Build: Completed



### ✔️ CI Completed
- ✅ Lint: Completed
- ✅ Build: Completed

hooni0918 committed:
feat/#56 - 앱 레이어 수정

hooni0918 committed:
feat/#56 앱 의존성 수정

hooni0918 committed:
feat/#56-DTO 생성

feat/#56 - DTO public으로 정의

hooni0918 committed:
feat/#56 데이터 모듈 및 레포지토리 구현체

feat/#56 - Utils 추가

feat/#56 - utils 의존

hooni0918 committed:
feat/#56 에러케이스 정의

chore/#56 네트워크 모듈 수정

hooni0918 committed:
feat/#56- 도메인로직 수정

hooni0918 committed:
feat/#56 불필요 코드 삭제

feat/#56 불필요 코드 삭제

feat/#56 불필요 코드수정

hooni0918 committed:
feat/#56 - 캘린더 뷰 수정

SijongKim93 commented on Projects/Core/NWNetwork/Sources/NWNetworkService.swift:
```diff
@@ -20,14 +20,22 @@ public class NWNetworkService: NWNetworkServiceProtocol {
+           let configuration = URLSessionConfiguration.default
+           configuration.httpAdditionalHeaders = headers
+           self.session = Session(configuration: configuration)
+       }
```
인증 헤더 넣어주다니 아주 감사용

SijongKim93 commented on Projects/Calendar/CalendarData/Sources/Repositories/CalendarRepositoryImpl.swift:
```diff
@@ -8,37 +8,38 @@
+    case serverError(String)
+    case networkError(String)
+    case unknown
+}
```
따로 네트워크 에러는 NWNetwork에 넣어뒀는데 중복될 것 같아요 한번 확인해보시구 NWNetwork 임포트했으니 그냥 새로 NetworkError 추가없이 사용해도 될 것 같습니다.

hooni0918 replied:
해결해씸더

SijongKim93 replied:
대씸더

hooni0918 committed:
fix/#56 컨플릭트 해결

hooni0918 committed:
fix/#56Simplifies project configurations

Removes redundant settings declarations in project configurations.
This streamlines the configuration process and reduces potential inconsistencies.

hooni0918 committed:
fix/#56 - Configures CalendarData module dependencies

Registers the CalendarRepository with a network service dependency in the DI container.
Removes the initial simple repository registration.
Deletes unused app dependency configuration.

hooni0918 committed:
fix/#56 - 컨플릭트는 해결
